assets/src/edit-story/components/canvas/useCanvasKeys.js

```diff
@@ -113,6 +113,17 @@ function useCanvasKeys(ref) {
             currentSelectedIds?.length === 1
               ? getNodeForElement(currentSelectedIds[0])
               : null;
+
+          // Check if the selection event happends outside the canvas container,
+          // i.e Checklist panel, Document panel... etc. If there is any text
+          // selection, we should not add the preventScroll setting, so that
+          // user can select the text properly.
+          if (global.getSelection().anchorOffset !== 0) {
```

Comment:
Are we sure this works?

If the selection begins with the first character in the `anchorNode`, `anchorOffset` returns `0`. So `0` is a legitimate value and does _not_ indicate that there isn't a selection.

In other words:

* No selection on page: `anchorOffset` returns `0`
* Selection(s) on page, starting at the beginning, `anchorOffset` can also return `0`

Reply:
Got it, so my understand is, we want to make sure the logic skip only when a `text object` is selected, then: 
1.  `anchorNode` is not null
2. `anchorOffset` is not 0

Reply:
Per my other comment, I don't see how `anchorNode` is ever `null`.

Please test again.

Reply:
Got it!
By doing some researches, both `sel.type` and `sel.toString()` are good ways to identify the selection type.
For `sel.type`:
If we select a text, the type should be `Range`; if we choose from the editor, the type will be `Caret`

For `sel.toString`:
If we select a text, the method should return the selected string; whereas from the editor, it will return an empty string.

I am thinking to go with the 2nd way and think the check will be enough.